Add support for multiple nem files

Kyle-Verhoog opened the pull request:
The closer the nem file is to the current directory the
higher precedence it has for finding, creating and deleting
entries.

p7g commented on nem.rs:
```diff
@@ -76,10 +133,32 @@ fn mnemonic(v: &[String], existing: Vec<&String>) -> String {
+    let mut path_buf = env::current_dir().expect("Couldn't get CWD");
+    loop {
+        path_buf.push(".nem.toml");
+        let path = path_buf.as_path();
```
I think you don't have to make a `Path` object from the `PathBuf`, most of the methods seem to accept `AsRef<Path>`, which a `PathBuf` implements.

p7g commented on nem.rs:
```diff
@@ -76,10 +133,32 @@ fn mnemonic(v: &[String], existing: Vec<&String>) -> String {
+                    nem_file.sort();
+                    nem_files.nem_files.push(nem_file);
+                }
+                Err(_) => {}
```
Maybe would be worth `eprintln!`-ing the error? Every `Error` in rust has to also implement `Display`, so you can format it with `"{}"`

p7g reviewed (approved):
:ok_hand: 

What if you're in a directory that isn't a child of the home directory? Should there be special cases like `~/.nem.toml` or `/etc/nem/nem.toml`?

Kyle-Verhoog commented on nem.rs:
```diff
@@ -62,6 +62,63 @@ struct Entry {
+    fn find_entry_by_code(&self, code: &String) -> Option<&Entry> {
+        self.nem_files
+            .iter()
+            .find(|nf| nf.find_entry_by_code(code).is_some())
+            .and_then(|nf| nf.find_entry_by_code(code))
```
yesss i was looking for a better way 🙌

Kyle-Verhoog committed:
implement multiple nem files

The closer the nem file is to the current directory the
higher precedence it has for finding, creating and deleting
entries.

Kyle-Verhoog committed:
add a test

Kyle-Verhoog commented:
Yes I believe the python implementation explicitly checked the home directory and ~/.config/.nem.toml. We should do a follow up with support for those and whatever other common config paths might exist 👍
